yarn-project/acir-simulator/src/client/execution_result.test.ts

```diff
@@ -0,0 +1,137 @@
+import { FunctionL2Logs } from '@aztec/types';
+import { ExecutionResult, collectEncryptedLogs } from './execution_result.js';
+import { PrivateCallStackItem } from '@aztec/circuits.js';
+
+describe('Execution Result test suite - collect encrypted logs', () => {
+  function emptyExecutionResultWithEncryptedLogs(encryptedLogs = FunctionL2Logs.empty()): ExecutionResult {
+    return {
+      acir: Buffer.from(''),
+      vk: Buffer.from(''),
+      partialWitness: new Map(),
+      callStackItem: PrivateCallStackItem.empty(),
+      readRequestCommitmentIndices: [],
+      preimages: {
+        newNotes: [],
+        nullifiedNotes: [],
+      },
+      returnValues: [],
+      nestedExecutions: [],
+      enqueuedPublicFunctionCalls: [],
+      encryptedLogs: encryptedLogs,
+      unencryptedLogs: FunctionL2Logs.empty(),
+    };
+  }
+
+  it('collect encrypted logs with nested fn calls', () => {
+    /*
+    Create the following executionResult object: 
+    fnA (log1) 
+        |---------->fnB (log2) 
+        |---------->fnC (log3) -> fnD (log4)
+        |---------->fnE (log5) 
+                     |-------->fnF (log6)
+                     |-------->fnG (log7) 
+    Circuits and ACVM process in a DFS + stack like format: [fnA, fnE, fnF, fnG fnC, fnD, fnB]
```

Comment:
```suggestion
    Circuits and ACVM process in a DFS + stack like format: [fnA, fnE, fnG, fnF fnC, fnD, fnB]
```
At least that's what I understand from the `yarn-project/aztec-rpc/src/kernel_prover/kernel_prover.ts`.

Reply:
yup you are right - in fact the test also returns that - forgot to write my comment correctly thanks!